### ISSUE TITLE
[JOSS] Issue in perceptron Python-binding documentation with mlpack class assignment

### ISSUE THREAD

**Issue by rasbt** [mlpack/mlpack#1419]
In the perceptron docs, at http://www.mlpack.org/docs/mlpack-3.0.1/python/perceptron.html

there are some issues in the code examples

```python
>>> from mlpack import perceptron

>>> perceptron(training=training_data, labels=training_labels)
>>> perceptron = output['output_model']

>>> perceptron(input_model=perceptron, test=test_data)
>>> predictions = output['output']
```

First there's the "output" issue referenced in #1418. For now, let's assume that's fixed:

```python
>>> output = perceptron(training=training_data, labels=training_labels)
>>> perceptron = output['output_model']

>>> output = perceptron(input_model=perceptron, test=test_data)
>>> predictions = output['output']
```

Now, the issue is this will raise an `TypeError: 'mlpack.perceptron.PerceptronModelType' object is not callable`. Presumably, because it's somehow trying to use the module `perceptron`. I would recommend not to overwrite the `perceptron` class imported from `mlpack` (earlier in the documentation) and modify it as follows (perceptron -> perceptron_model):

```python
>>> output = perceptron(training=training_data, labels=training_labels)
>>> perceptron_model = output['output_model']

>>> output = perceptron(input_model=perceptron_model, test=test_data)
>>> predictions = output['output']
```

Would probably be a good idea to double-check the other modules as well

**Comment by rcurtin** [comment]
Good call---I have a fix in #1421.  This is one of the tricky bits with writing agnostic documentation: for the command-line, this makes sense.  The model just gets called `perceptron.bin`.  But in Python it goes south fast like you pointed out. :)

I checked all the other bindings too, and none have the same issue.

**Comment by rcurtin** [comment]
Fixed with the merge of #1421.
